src/components/tutorial-card/helpers/with-auth-elements.tsx

```diff
@@ -0,0 +1,65 @@
+import { AUTH_ENABLED } from 'hooks/use-authentication'
+import { useTutorialProgress } from 'hooks/progress'
+import {
+	progressStatusToPercent,
+	progressStatusToAriaLabel,
+} from 'lib/learn-client/api/progress'
+import { TutorialProgressStatus } from 'lib/learn-client/types'
+import { TutorialCardBookmarkButton } from 'components/bookmark-button'
+import TutorialCard, { TutorialCardPropsWithId } from '..'
+import ProgressIconAndLabel from '../components/progress-icon-and-label'
+import { getSpeakableDuration } from './build-aria-label'
+
+/**
+ * Displays a TutorialCard, which shows additional user-data-specific elements
+ * for authenticated users.
+ */
+export function TutorialCardWithAuthElements(props: TutorialCardPropsWithId) {
+	const { id: tutorialId, collectionId, ...rest } = props
+
+	/**
+	 * Get tutorial progress. Will be undefined if not authenticated.
+	 * Note as well that useTutorialProgress depends on AUTH_ENABLED.
+	 */
+	const { tutorialProgressStatus } = useTutorialProgress({
+		tutorialId,
+		collectionId,
+	})
+
+	/**
+	 * Determine whether to render progress status.
+	 * We only show progress for at a minimum state of `in_progress`.
+	 */
+	const isProgressLoaded = typeof tutorialProgressStatus !== 'undefined'
+	const meetsProgressThreshold =
+		parseInt(progressStatusToPercent(tutorialProgressStatus)) >=
+		parseInt(progressStatusToPercent(TutorialProgressStatus.in_progress))
+	const shouldRenderProgress = isProgressLoaded && meetsProgressThreshold
```

Comment:
nit suggestion: what do you think about this? It felt clearer to me looking at this to use the status enum instead of coercing it to a number. We could also pull out the "display progress statuses" into a shared constant if we need to check elsewhere.

```suggestion
  const meetsProgressThreshold = [TutorialProgressStatus.in_progress, TutorialProgressStatus.complete].includes(tutorialProgressStatus)
	const shouldRenderProgress = isProgressLoaded && meetsProgressThreshold
```

Reply:
I dig it! 👍

I was very on the fence... part of me is wondering whether it might make sense to keep the progress data in `percent` for longer, so that progress comparison things are more intuitive... Then we'd only convert the `progressPercentToStatus` right before we display it 🤔 The enums made the most sense to me at first since they were easiest to map to the implementation... but that definitely  seems a candidate for some refactoring 👍